src/utils/CometAdapter.cpp

```diff
@@ -0,0 +1,589 @@
+    registerIntOption_("spectrum_batch_size", "<num>", 0, "max. // of spectra to search at a time; 0 to search the entire scan range in one loop", false, true);
+    registerDoubleOption_("mass_offsets", "<offset>", 0, "one or more mass offsets to search (values substracted from deconvoluted precursor mass)", false, true);
+
+    // spectral processing
+    registerIntOption_("minimum_peaks", "<num>", 10, "required minimum number of peaks in spectrum to search (default 10)", false, true);
+    registerIntOption_("minimum_intensity", "<num>", 0, "minimum intensity value to read in", false, true);
+    registerIntOption_("remove_precursor_peak", "<num>", 0, "0=no, 1=yes, 2=all charge reduced precursor peaks (for ETD)", false, true);
+    registerIntOption_("remove_precursor_tolerance", "<num>", 1.5, "+- Da tolerance for precursor removal", false, true);
+    registerStringOption_("clear_mz_range", "[min] [max]", "0.0 0.0", "for iTRAQ/TMT type data; will clear out all peaks in the specified m/z range", false, true);
+
+    //Modifications
+    registerStringList_("fixed_modifications", "<mods>", ListUtils::create<String>(""), "Fixed modifications, specified using UniMod (www.unimod.org) terms, e.g. 'Carbamidomethyl (C)' or 'Oxidation (M)'", false, false);
+    vector<String> all_mods;
+    ModificationsDB::getInstance()->getAllSearchModifications(all_mods);
+    setValidStrings_("fixed_modifications", all_mods);
+    registerStringList_("variable_modifications", "<mods>", ListUtils::create<String>(""), "Variable modifications, specified using UniMod (www.unimod.org) terms, e.g. 'Carbamidomethyl (C)' or 'Oxidation (M)'", false, false);
+    setValidStrings_("variable_modifications", all_mods);
+    registerIntOption_("max_variable_mods_in_peptide", "<num>", 5, "", false, true);
+  }
+
+  vector<ResidueModification> getModifications_(StringList modNames)
+  {
+    vector<ResidueModification> modifications;
+
+    // iterate over modification names and add to vector
+    for (StringList::iterator mod_it = modNames.begin(); mod_it != modNames.end(); ++mod_it)
+    {
+      String modification(*mod_it);
+      modifications.push_back(ModificationsDB::getInstance()->getModification(modification));
+    }
+
+    return modifications;
+  }
+
+  void removeTempDir_(const String& tmp_dir)
+  {
+    if (tmp_dir.empty()) return; // no temp. dir. created
+
+    if (debug_level_ >= 2)
+    {
+      writeDebug_("Keeping temporary files in directory '" + tmp_dir + "'. Set debug level to 1 or lower to remove them.", 2);
+    }
+    else
+    {
+      if (debug_level_ == 1) writeDebug_("Deleting temporary directory '" + tmp_dir + "'. Set debug level to 2 or higher to keep it.", 1);
+      File::removeDirRecursively(tmp_dir);
+    }
+  }
+
+  void createParamFile_(ostream& os)
+  {
+    os << "# comet_version 2016.01 rev. 2\n";               //required as first line in the param file
```

Comment:
I just tried the adapter with Comet version "2017.01 rev. 0beta" and got this error:
> The comet.params file is from version 2016.01
> Please update your comet.params file.  You can generate
> a new parameters file using "comet -p"

So unless we want to tie the adapter to one specific version of Comet, it would be necessary to have a parameter (probably advanced) that allows to set the version that gets written in this line.